include/soci/session.h

```diff
@@ -215,6 +217,8 @@ class SOCI_DECL session
     SOCI_NOT_COPYABLE(session)
 
     void reset_after_move();
+    struct schema_table_name * alloc_schema_table_name(std::string & tableName);
```

Comment:
2 comments here, one minor but the other one important:

1. `struct` is redundant and should be removed.
2. Is the returned pointer owned by the caller? If so, it should be returned as `std::unique_ptr<schema_table_name>`.

Reply:
The schema_table_name is only on loan. It is stored in the linked list commented on elsewhere as it must be stored for the lifetime of the prepare_temp_type object - so deleting it when exiting the scope of the method will break the intended functionality.

Reply:
Please add a comment saying that this pointer is owned by the class and shouldn't be deleted then, as this is rather non-obvious.

Alternatively, and even better, if the returned pointer is always non-null, we could return a reference here instead.

Reply:
References are so much better - it's coming in a few minutes - when I've tested it locally.